src/ioc/index.ts

```diff
@@ -1,12 +1,16 @@
 import IoCContainer from 'ioc-lite';
 
-// import { Logger } from '../services/logger';
-// import { HTTP } from '../services/http';
-// import { Users } from '../services/users';
+import { Logger } from '../services/logger';
+import { HTTP } from '../services/http';
+import { Users } from '../services/users';
 
+// using a generic type allows you to control which resources can be registered.
 export const createIoCContainer = () =>  {
   const ioc = new IoCContainer();
```

Comment:
Please use a generic type to allow TS to check the type of register resources.
I added this [comment](https://github.com/kvasenin/Dependencies-Injection/commit/b9b6c195dbce38268fc0435faeb1cac8ed20b7d6#r116899251) not to make you remove the type, but to use it.